lib/web/apiserver.go

```diff
@@ -2148,15 +2149,26 @@ func clientMetaFromReq(r *http.Request) *auth.ForwardedClientMetadata {
 	}
 }
 
-// deleteWebSession is called to sign out user
+// deleteWebSession is called to sign out user from web, app and SAML IdP session.
 //
 // DELETE /v1/webapi/sessions/:sid
 //
 // Response:
 //
 // {"message": "ok"}
 func (h *Handler) deleteWebSession(w http.ResponseWriter, r *http.Request, _ httprouter.Params, ctx *SessionContext) (interface{}, error) {
-	err := h.logout(r.Context(), w, ctx)
+	// TODO(sshah): we can skip this step below once we have a mechanism to update websession
+	// with SAML details.
+	samlSessionCookie, err := r.Cookie(samlidp.SAMLSessionCookieName)
+	if err != nil {
+		h.log.Debug("SAML IdP session not found for deletion for user: %s", ctx.GetUser())
```

Comment:
How unexpected is this condition? A comment clarifying would help.

Reply:
My initial plan was to swallow the error completely. For generic Teleport installation, this is just going to be log noise, I've removed this in https://github.com/gravitational/teleport/pull/37619/commits/7f451952eb724dda594c4e7e45eb8746b0d35e13